src/common/function/src/scalars/matches.rs

```diff
@@ -980,18 +1136,21 @@ mod test {
                             op: UnaryOp::Optional,
                             pattern: "a".to_string(),
                         },
-                        PatternAst::Binary {
-                            op: BinaryOp::Or,
-                            children: vec![
-                                PatternAst::Literal {
-                                    op: UnaryOp::Optional,
-                                    pattern: "b".to_string(),
-                                },
-                                PatternAst::Literal {
-                                    op: UnaryOp::Optional,
-                                    pattern: "c".to_string(),
-                                },
-                            ],
+                        PatternAst::Group {
+                            op: UnaryOp::Optional,
```

Comment:
Can we add some tests for groups with `Must` or `Negative` op? Now we only have tests for groups with `Optional` op.

Reply:
`evaluate_matches` has several cases like `+(fox jumps) AND over` or `over -(fox jumps)`